code/game/objects/items/pneumaticCannon.dm

```diff
@@ -205,6 +205,16 @@
 	var/turf/newtarget = locate(new_x, new_y, starting.z)
 	return newtarget
 
+/obj/item/pneumatic_cannon/handle_atom_del(atom/A)
+	. = ..()
+	if (A in loadedItems)
+		var/obj/item/I = A
+		loadedItems -= I
```

Comment:
Just do `loadedItems -= I` since both `in` and `-=` have to iterate it anyway. Use .len to check if it changed

Reply:
Or, way up the cost of ```-=``` and a ```.len```, vs a single ```.Remove()```

```
if(loadedItems.Remove(A))
    var/obj/item/I = A
    loadedWeightClass -= I.w_class
```

Reply:
yeah that's actually better